components/de_amp/browser/test/de_amp_browsertest.cc

```diff
@@ -169,14 +211,42 @@ IN_PROC_BROWSER_TEST_F(DeAmpBrowserTest, SimpleDeAmp) {
   // Now go to an AMP page
   https_server_.reset(new net::EmbeddedTestServer(
       net::test_server::EmbeddedTestServer::TYPE_HTTPS));
-  https_server_->RegisterRequestHandler(base::BindRepeating(
-      HandleRequest, kTestHost, kTestCanonicalPage, kTestAmpBody));
+  https_server_->RegisterRequestHandler(
+      base::BindRepeating(HandleRequest, kTestCanonicalPage, kTestAmpBody));
   ASSERT_TRUE(https_server_->Start());
   const GURL original_url = https_server_->GetURL(kTestHost, kTestAmpPage);
   const GURL landing_url = https_server_->GetURL(kTestHost, kTestCanonicalPage);
   NavigateToURLAndWaitForRedirects(original_url, landing_url);
 }
 
+IN_PROC_BROWSER_TEST_F(DeAmpBrowserTest, AmpPagesPointingAtEachOther) {
+  TogglePref(true);
+  https_server_->RegisterRequestHandler(
+      base::BindRepeating(HandleRequestForRedirectTest, kTestAmpBody));
+  ASSERT_TRUE(https_server_->Start());
+  const GURL original_url =
+      https_server_->GetURL(kTestHost, kTestRedirectingAmpPage1);
+  const GURL landing_url =
+      https_server_->GetURL(kTestHost, kTestRedirectingAmpPage2);
+  NavigateToURLAndWaitForRedirects(original_url, landing_url);
+}
+
+IN_PROC_BROWSER_TEST_F(DeAmpBrowserTest, CanonicalRedirectsToAmp) {
```

Comment:
this needs to start from the original amp page to properly replicate the issue as mentioned  in DM and on the issue